book/src/05_storages.md

```diff
@@ -0,0 +1,78 @@
+# Storages
+
+Specs contains a bunch of different storages, all built and optimized for
+different use cases. But let's see some basics first.
+
+## Storage basics
+
+What you specify in a system `impl`-block is an `UnprotectedStorage`.
+Each `UnprotectedStorage` exposes an unsafe getter which does not
+perform any checks whether the requested index for the component is valid
+(the id of an entity is the index of its component). To allow checking them
+and speeding up iteration, we have something called hierarchical bitsets,
+provided by [`hibitset`](https://github.com/slide-rs/hibitset).
+
+> **Note:** In case you don't know anything about bitsets,
+  you can safely skip the following section about it. Just keep
+  in mind that we have some mask which tracks for
+  which entities a component exists.
+
+How does it speed up the iteration? A hierarchical bitset is essentially
+a multi-layer bitset, where each upper layer "summarizes" multiple bits
+of the underlying layers. That means as soon as one of the underlying
+bits is `1`, the upper one also becomes `1`, so that we can skip a whole
+range of indices if an upper bit is `0` in that section. In case it's `1`,
+we go down by one layer and perform the same steps again (it currently
+has 4 layers).
+
+## Storage overview
+
+Here a list of the storages with a short description and a link
+to the corresponding heading.
+
+|Storage Type         |Description               |Optimized for                 |
+|:-------------------:|--------------------------|------------------------------|
+| [`BTreeStorage`]    | Works with a `BTreeMap`  | no particular case           |
+| [`DenseVecStorage`] | Uses a redirection table | fairly often used components |
+| [`HashMapStorage`]  | Uses a `HashMap`         | rare components              |
+| [`NullStorage`]     | Can flag entities        | universally applicable       |
```

Comment:
I find "universally applicable" a bit confusing, given that it's not really an alternative to others